src/main/java/com/example/userrestapi/controller/UserController.java

```diff
@@ -23,20 +27,37 @@
 
         List<User> users = new ArrayList<User>();
```

Comment:
```suggestion
        List<User> users;
```
the initialization here is not needed as you overwrite that just after